R/rvw_example/dt2vw.R

```diff
@@ -1,251 +0,0 @@
-# Format data.table to Vowpal wabbit format
-# @param data[data.table]: data.table format (to be transformed)
-# @param fileName[string]:  file name of the resulting data in VW-friendly format
-# @param namespaces[list / yaml file]: name of each namespace and each variable for each namespace can be a R list, or a YAML file
-# @param target[string]: target of the data (target)
-# @param weight[string]: weight of each line of the dataset (importance)
-# @param tag[string]: tag of each line of the dataset
-# @param hard_parse[bool]: if equals true, parses the data more strictly to avoid feeding VW with false categorical
-# @param variables like '_', or same variables perceived differently life "_var" and "var"
-# examples
-# library(ggplot2)
-# library(handy)
-# library(data.table)
-# dt = diamonds
-# head(dt)
-# data_types = handy::get_feature_type(dt, threshold = 50)
-# namespaces = list(n = list(varName = data_types$num_vars, keepSpace=F),
-#                  c = list(varName = data_types$fact_vars, keepSpace=F))
-# dt = setDT(dt)
-# dt$y = with(dt, ifelse(y < 5.71, 1, -1))
-# dt[, weight:=ifelse(y == 1, 1.0, 3.0345)]
-# 
-# # Use the function directly if the file is small
-# dt2vw(dt, '../sample_diamonds2.vw', namespaces, target='y', weight='weight')
-# system('head ../sample_diamonds2.vw')
-# 
-# inxs = c(0, seq(1000, 50000, 1000), nrow(dt))
-# i = 0
-# while(i < (length(inxs) - 1) ){
-#   i = i + 1
-#   start = inxs[i] + 1
-#   end = inxs[(i+1)]
-#   print(start)
-#   print(end)
-#   dt2vw(dt=train[start:end], fileName = '../out_in_chunks.vw',
-#         namespaces = nmp, target = 'target', weight=NULL, append=TRUE)
-# }
-dt2vw <- function(data, fileName, namespaces = NULL, target, weight = NULL,
-                  tag = NULL, hard_parse = F, append = F){
-  #required packages
-  require(data.table)
-  
-  if(class(data)[1] != "data.table")
-    data = data.table(data)
-  
-  #change target if its boolean to take values in {-1,1}
-  if(is.logical(data[[target]]) | sum(levels(factor(data[[target]])) == levels(factor(c(0,1)))) == 2)
-  {
-    data[[target]][data[[target]] == TRUE] = 1
```

Comment:
using `=` to change data will force a copy of the whole data.table.  Try the [`data.table::set`](http://www.inside-r.org/packages/cran/data.table/docs/set) function instead.